ecosystem/typescript/sdk/src/providers/indexer.ts

```diff
@@ -310,4 +318,141 @@ export class IndexerClient {
+    IndexerClient.validateAddress(address);
+    const graphqlQuery = {
+      query: GetOwnedTokens,
+      variables: { address, offset: extraArgs?.options?.offset, limit: extraArgs?.options?.limit },
+    };
+    return this.queryIndexer(graphqlQuery);
+  }
+
+  /**
+   * Queries all tokens of a specific collection that an account owns by the collection address
+   *
+   * @param ownerAddress owner address that owns the tokens
+   * @param collectionAddress the collection address
+   * @returns GetTokenOwnedFromCollectionQuery response type
+   */
+  async getTokenOwnedFromCollectionAddress(
+    ownerAddress: MaybeHexString,
+    collectionAddress: string,
+    extraArgs?: {
+      tokenStandard?: TokenStandard;
+      options?: PaginationArgs;
+    },
+  ): Promise<GetTokenOwnedFromCollectionQuery> {
+    const ownerHexAddress = HexString.ensure(ownerAddress).hex();
+    IndexerClient.validateAddress(ownerHexAddress);
+
+    const collectionHexAddress = HexString.ensure(collectionAddress).hex();
+    IndexerClient.validateAddress(collectionHexAddress);
+
+    const graphqlQuery = {
+      query: GetTokenOwnedFromCollection,
+      variables: {
+        collection_id: collectionHexAddress,
+        owner_address: ownerHexAddress,
+        offset: extraArgs?.options?.offset,
+        limit: extraArgs?.options?.limit,
+      },
+    };
+    return this.queryIndexer(graphqlQuery);
+  }
+
+  /**
+   * Queries all tokens of a specific collection that an account owns by the collection name and collection
+   * creator address
+   *
+   * @param ownerAddress owner address that owns the tokens
+   * @param collectionName the collection name
+   * @param creatorAddress the collection creator address
+   * @returns GetTokenOwnedFromCollectionQuery response type
+   */
+  async getTokenOwnedFromCollectionNameAndCreatorAddress(
+    ownerAddress: MaybeHexString,
+    collectionName: string,
+    creatorAddress: MaybeHexString,
+    extraArgs?: {
+      tokenStandard?: TokenStandard;
+      options?: PaginationArgs;
+    },
+  ): Promise<GetTokenOwnedFromCollectionQuery> {
+    const collectionAddress = await this.getCollectionAddress(creatorAddress, collectionName, extraArgs);
+    const tokens = await this.getTokenOwnedFromCollectionAddress(ownerAddress, collectionAddress, extraArgs);
+    return tokens;
+  }
+
+  /**
+   * Queries data of a specific collection by the collection creator address and the collection name.
+   *
+   * if, for some reason, a creator account has 2 collections with the same name in v1 and v2,
+   * can pass an optional `tokenStandard` parameter to query a specific standard
+   *
+   * @param creatorAddress the collection creator address
+   * @param collectionName the collection name
+   * @returns GetCollectionDataQuery response type
+   */
+  async getCollectionData(
+    creatorAddress: MaybeHexString,
+    collectionName: string,
+    extraArgs?: {
+      tokenStandard?: TokenStandard;
```

Comment:
hm, does this mean that there could be 2 different standard from the same collection? For example, AptosMonkey v1, and AptosMonkey v2 collection?

Reply:
it is not the same collection, these are 2 different collection that were created with the same account and have the same name. 
for [example](https://cloud.hasura.io/public/graphiql?endpoint=https%3A%2F%2Findexer-testnet.staging.gcp.aptosdev.com%2Fv1%2Fgraphql&query=query+TokenOwnership+%7B%0A++current_token_ownerships_v2%28%0A++++where%3A+%7Bowner_address%3A+%7B_eq%3A+%220x6ed17703663230cd3876400faaeb2ab71c4aa8812b625823db0c236ab5fa98b1%22%7D%2C+current_token_data%3A+%7Btoken_name%3A+%7B_eq%3A+%22Alice+Token%22%7D%7D%7D%0A++%29+%7B%0A++++token_standard%0A++++is_fungible_v2%0A++++is_soulbound_v2%0A++++is_soulbound_v2%0A++++property_version_v1%0A++++table_type_v1%0A++++token_properties_mutated_v1%0A++++amount%0A++++last_transaction_timestamp%0A++++last_transaction_version%0A++++storage_id%0A++++owner_address%0A++++current_token_data+%7B%0A++++++token_name%0A++++++token_data_id%0A++++++token_uri%0A++++++token_properties%0A++++++supply%0A++++++maximum%0A++++++last_transaction_version%0A++++++last_transaction_timestamp%0A++++++largest_property_version_v1%0A++++++current_collection+%7B%0A++++++++collection_name%0A++++++++creator_address%0A++++++++description%0A++++++++uri%0A++++++++collection_id%0A++++++++last_transaction_version%0A++++++++current_supply%0A++++++++mutable_description%0A++++++++total_minted_v2%0A++++++++table_handle_v1%0A++++++++mutable_uri%0A++++++%7D%0A++++%7D%0A++%7D%0A++current_collections_v2%28%0A++++where%3A+%7Bcollection_name%3A+%7B_eq%3A+%22AliceCollection%22%7D%2C+creator_address%3A+%7B_eq%3A+%220x6ed17703663230cd3876400faaeb2ab71c4aa8812b625823db0c236ab5fa98b1%22%7D%7D%0A++%29+%7B%0A++++collection_id%0A++++token_standard%0A++++collection_name%0A++++creator_address%0A++++current_supply%0A++++description%0A++++uri%0A++%7D%0A%7D%0A)